### PR TITLE
9933 - fix twisted.internet.process.pauseProducing/resumeProducing fo…

### PR DESCRIPTION
**Pull request by ndg63276**
…r python 3

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/9933
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

**Review by auvipy — APPROVED** [review]

File: src/twisted/internet/test/test_process.py [hunk 1]
```diff
@@ -843,6 +843,26 @@ def start():
...
+        """
+
+        def pauseAndResume(reactor):
+            try:
+                protocol = ProcessProtocol()
+                transport = reactor.spawnProcess(
+                    protocol, pyExe, [pyExe, b"-c", b""], usePTY=self.usePTY
+                )
+                transport.pauseProducing()
+                transport.resumeProducing()
```
Comment (graingert) [inline src/twisted/internet/test/test_process.py:858]: we should test this streaming to an object with actual flow control

Comment (graingert) [reply src/twisted/internet/test/test_process.py:858]: here's a test using a MemoryProcessReactor that could be updated to use a real reactor

https://github.com/twisted/twisted/blob/c6c863bf8430f5f1e017af874ffdc89b2cc1b710/src/twisted/internet/test/test_endpoints.py#L822-L1140

Comment (exarkun) [reply src/twisted/internet/test/test_process.py:858]: I'm not totally clear about this review. Is this a request to change the test in the PR?

Comment (graingert) [reply src/twisted/internet/test/test_process.py:858]: To add an additional test that uses a real reactor and process

Comment (exarkun) [reply src/twisted/internet/test/test_process.py:858]: I think the test in the PR does use a real reactor and a real process already.  It uses the "reactor builder" testing framework which runs the test against *all* of the real reactors, even.

Comment (graingert) [reply src/twisted/internet/test/test_process.py:858]: ah ok, that makes sense

**Review by graingert — APPROVED** [review]
This looks like a good patch and the test does cover the error here.

I'd like to see a test that checked the side effect of pause/resumeProducing as currently it's just a smoke test

**Review by wiml — APPROVED** [review]
This looks good to me.

I largely agree with @graingert  that it would be nice for the test to verify that `pauseProducing`/`resumeProducing` in fact stop-and-start producing data, but presumably the underlying `stopReading`/`startReading` calls are already tested in the reactor tests (and it doesn't seem trivia to add that here), so I don't think that should block merging this PR.

**Pull request merged**